CMakeLists.txt

```diff
@@ -0,0 +1,121 @@
+cmake_minimum_required(VERSION 3.19)
+
+project(finufft VERSION 2.0.4 LANGUAGES C CXX)
+
+set(GNU_LIKE_FRONTENDS AppleClang Clang GNU)
+if(CMAKE_CXX_COMPILER_ID IN_LIST GNU_LIKE_FRONTENDS)
+    # Set custom compiler flags for gcc-compatible compilers
+    set(CMAKE_CXX_FLAGS_RELEASE "-O3 -DNDEBUG -funroll-loops")
+    set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O3 -g -DNDEBUG -funroll-loops")
+endif()
+
+include(CTest)
+
+set(FINUFFT_ARCH_FLAGS "-march=native" CACHE STRING "Compiler flags for specifying target architecture.")
+set(FINUFFT_FFTW_SUFFIX "OpenMP" CACHE STRING "Suffix for FFTW libraries (e.g. OpenMP, Threads etc.)")
+set(FINUFFT_BUILD_EXAMPLES "OFF" CACHE BOOL "Whether to build the FINUFFT examples")
+set(FINUFFT_BUILD_TESTS "OFF" CACHE BOOL "Whether to build the FINUFFT tests")
+set(FINUFFT_BUILD_FORTRAN "OFF" CACHE BOOL "Whether to build the FINUFFT Fortran examples")
+set(FINUFFT_BUILD_MATLAB "OFF" CACHE BOOL "Whether to build the FINUFFT Matlab interface")
+set(FINUFFT_ENABLE_SANITIZERS "ON" CACHE BOOL "Whether to enable sanitizers, only effective for Debug configuration.")
+set(FINUFFT_USE_OPENMP "ON" CACHE BOOL "Whether to use OpenMP for parallelization. If disabled, the finufft library will be single threaded. This does not affect the choice of FFTW library.")
+
+if(FINUFFT_BUILD_MATLAB)
+    # When building for matlab, we will fetch the OpenMP library used by matlab
+    # instead of system default for compatibility.
+    find_package(Matlab REQUIRED)
+    find_library(OpenMP_iomp5_LIBRARY NAMES iomp5 HINTS ${Matlab_ROOT_DIR}/sys/os/ PATH_SUFFIXES glnxa64 maci64)
```

Comment:
I like this idea. But I see how the name `iomp5` is then only used by the `matlab` preset in the json file.  I didn't use this. Instead I did `cmake .. -DFINUFFT_BUILD_MATLAB` and foudn that it linked to `-gomp`.  THis is fine for linux, not for mac.  But I'm wondering why matlab should be a separate preset rather than a flag option in the standard cmake build?

Reply:
Good point - I've moved more of the logic into the `CMakeLists.txt` file and changed the solution a little bit (went from very hacky to a bit less hacky).